src/main/java/org/lightcouch/View.java

```diff
@@ -373,7 +373,19 @@ public boolean queryForBoolean() {
         final JsonObject currentKeys = new JsonObject();
         final JsonObject jsonNext = new JsonObject();
         final JsonObject jsonPrev = new JsonObject();
-        currentKeys.addProperty(CURRENT_START_KEY, rows.get(0).getKey());
+
+        Object getKey = rows.get(0).getKey();
+        //Handle multi value key array
+        if(getKey instanceof ArrayList) {
+            String keyAsString = "[";
+            for(Object key : (ArrayList<Object>)getKey) {
+                keyAsString += key.toString() + ",";
+            }
+            //Final key value as string with closing bracket
+            getKey = keyAsString.substring(0,keyAsString.length()-1).concat("]");
+        }
```

Comment:
Yeah it would be cleaner if it can be a JsonArray rather than trying to manipulate the string